internal/vppinit/apiparams/apiparams.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2024 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package apiparams provides parsing factilty for configuration parameters
+// file for vpp init
+package apiparams
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v3"
+
+	"github.com/networkservicemesh/govpp/binapi/af_packet"
+	"github.com/networkservicemesh/govpp/binapi/af_xdp"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+)
+
+// ConfigYAML contains parameters for various AF types
+type ConfigYAML struct {
```

Comment:
Could be ranamed if we move this file to the vppinit package.

```suggestion
type VPPInitParameters struct {
```

Reply:
Renamed this struct to 'Parameters' since the static checker doesn't like the 'VPPInitParameters'

```
internal/vppinit/apiparams.go:35:6: exported: type name will be used as vppinit.VPPInitParameters by other packages, and that stutters; consider calling this Parameters (revive)
type VPPInitParameters struct {
```